internal/service/team/update_user.go

```diff
@@ -0,0 +1,89 @@
+package team
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20240805004/admin"
+)
+
+func UpdateTeamUsers(teamsAPI admin.TeamsApi, usersAPI admin.MongoDBCloudUsersApi, existingTeamUsers *admin.PaginatedApiAppUser, newUsernames []string, orgID, teamID string) error {
```

Comment:
should we check that at least one user remains or we leave the error to Atlas?

Reply:
would leave the error to atlas so implementation is flexible to future changes (e.g. API deciding to support empty teams again)